### PR TITLE
Implemented: code to add timezone support on the settings page (#2a7ymej).

### PR DESCRIPTION
**Pull request by meet-aniket**
### Related Issues

Closes #

### Short Description and Why It's Useful

 Implemented functionality to add support to change user timezone from settings page of the app.

### Screenshots of Visual Changes before/after (If There Are Any)

https://user-images.githubusercontent.com/52008359/167565107-8c2b7894-b686-485f-a649-9755c6ce247b.mp4

**IMPORTANT NOTICE** - Remember to add a changelog entry

### CODE REVIEW

File: src/components/SaveThresholdModal.vue [hunk 1]
```diff
@@ -187,7 +187,7 @@ export default defineComponent({
           'maxRecurrenceCount': '-1',
           'parentJobId': job.parentJobId,
           'runAsUser': 'system', // default system, but empty in run now
-          'recurrenceTimeZone': DateTime.now().zoneName
+          'recurrenceTimeZone': this.store.state.user.current.userTimeZone
```
Comment (ymaheshwari1) [inline src/components/SaveThresholdModal.vue:190]: Use getter to fetch the timezone.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/TimezoneModal.vue [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        message,
+        buttons: [
+            {
+              text: this.$t("Cancel"),
+            },
+            {
+              text: this.$t("Confirm"),
+              handler: () => {
+                this.setUserTimeZone();
+                }
```
Comment (ymaheshwari1) [inline src/views/TimezoneModal.vue:95]: Improve indentation.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/SaveThresholdModal.vue (hunk 1 above)
Comment (meet-aniket) [reply src/components/SaveThresholdModal.vue]: Sir, update code.

File: src/views/TimezoneModal.vue (hunk 2 above)
Comment (meet-aniket) [reply src/views/TimezoneModal.vue]: Sir, improved code.

File: src/components/JobConfiguration.vue [hunk 3]
```diff
@@ -239,11 +240,11 @@ export default defineComponent({
       }
     },
     getTime (time: any) {
-      return DateTime.fromMillis(time).toLocaleString(DateTime.DATETIME_MED);
+      return DateTime.fromMillis(time, { zone: this.getUserProfile.userTimeZone }).toLocaleString(DateTime.TIME_SIMPLE);
```
Comment (ymaheshwari1) [inline src/components/JobConfiguration.vue:243]:
```suggestion
      return DateTime.fromMillis(time, { zone: this.getUserProfile.userTimeZone }).toLocaleString(DateTime.DATETIME_MED);
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/JobConfiguration.vue (hunk 3 above)
Comment (meet-aniket) [reply src/components/JobConfiguration.vue]: Improved code Sir.

File: src/components/JobConfiguration.vue [hunk 4]
```diff
@@ -125,7 +125,8 @@ export default defineComponent({
       getJobStatus: 'job/getJobStatus',
       getJob: 'job/getJob',
       shopifyConfigId: 'user/getShopifyConfigId',
-      currentEComStore: 'user/getCurrentEComStore'
+      currentEComStore: 'user/getCurrentEComStore',
+      getUserProfile: 'user/getUserProfile'
```
Comment (adityasharma7) [inline src/components/JobConfiguration.vue:129]:
```suggestion
      userProfile: 'user/getUserProfile'
```

File: src/components/JobHistoryModal.vue [hunk 5]
```diff
@@ -76,18 +76,19 @@ export default defineComponent({
     ...mapGetters({
       getEnumName: 'job/getEnumName',
       getCurrentEComStore:'user/getCurrentEComStore',
-      getStatusDesc: 'util/getStatusDesc'
+      getStatusDesc: 'util/getStatusDesc',
+      getUserProfile: 'user/getUserProfile'
```
Comment (adityasharma7) [inline src/components/JobHistoryModal.vue:80]:
```suggestion
      userProfile: 'user/getUserProfile'
```

File: src/views/TimezoneModal.vue [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    <ion-toolbar>
+      <ion-buttons slot="start">
+        <ion-button @click="closeModal"> 
+          <ion-icon :icon="close" />
+        </ion-button>
+      </ion-buttons>
+      <ion-title>{{ $t("Select time zone") }}</ion-title>
+    </ion-toolbar>
+    <ion-toolbar>
+      <ion-searchbar @ionFocus="selectSearchBarText($event)" :placeholder="$t('Search time zones')"  v-model="queryString" @ionInput="findTimeZone()" />
```
Comment (adityasharma7) [inline src/views/TimezoneModal.vue:12]: This should work enter

Comment (adityasharma7) [reply src/views/TimezoneModal.vue:12]: Also, we should handle text mismatch

File: src/views/TimezoneModal.vue [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+      this.filteredTimeZones = this.timeZones.filter((timeZone: any) => {
+        return regularExp.test(timeZone.id) || regularExp.test(timeZone.label);
+      });
+    },
+    async getAvailableTimeZones() {
+      UserService.getAvailableTimeZones().then((resp: any) => {
+        if(resp.status === 200 && !hasError(resp)) {
+          this.timeZones = resp.data.filter((timeZone: any) => {
+            return DateTime.local().setZone(timeZone.id).isValid;
```
Comment (adityasharma7) [inline src/views/TimezoneModal.vue:135]: Add comment why we are doing this?

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/JobConfiguration.vue (hunk 4 above)
Comment (meet-aniket) [reply src/components/JobConfiguration.vue:129]: Sir, updated code.

File: src/components/JobHistoryModal.vue (hunk 5 above)
Comment (meet-aniket) [reply src/components/JobHistoryModal.vue:80]: Sir, updated code.

File: src/views/TimezoneModal.vue (hunk 6 above)
Comment (meet-aniket) [reply src/views/TimezoneModal.vue]: Sir, updated code.

**Pull request merged**